### PR TITLE
wrap LSTMCell in DropoutWrapper if dropout param

### PR DESCRIPTION
**Pull request by lefnire**
Adds a `dropout=_` param to `lstm()`, which if provided wraps `lstm_cell` in a `rnn.DropoutWrapper` for regularization. Feel free to close if there's a smarter way, more for consideration on providing a dropout option for these layers where applicable.

### CODE REVIEW

**Comment by michaelschaarschmidt** [comment]
Hey thanks for this,

we will have a think and merge/close

**Comment by AlexKuhnle** [comment]
Thanks for the PR, looks all good! :+1:
